### ISSUE TITLE
Checklist Field - Param to adjust columns and more

### ISSUE THREAD

**Issue by PiahDoNeumann** [Laravel-Backpack/CRUD#2803]
In the checklist field on line 32, it has the class "col-sm-4" and I would like to change it to "col-sm-6", to allow only two columns or also "col-sm-3" for 4 columns of data. Would you be able to include a parameter to set one of the classes and not need to open a new field?

What about the possibility of including a button for "select" or "deselect" in the "checklist" field, so that the user does not have to select one by one?

**Comment by tabacitu** [comment]
Both of them sound like cool features, I agree! Thanks @PiahDoNeumann . 

I've added this to the 4.1 board, and marked it as a "could have" - so we won't start working on this ASAP I'm afraid (our priorities are Must, then Should, only afterwards Could). If you need this very soon, and end up working out a solution yourself, please consider submitting a PR - it would make it faster for us to offer this to everyone. Especially the custom column is a pretty simple change - as simple as using ```{{ $field['option_class'] ?? 'col-sm-4' }}```

Cheers!

**Comment by PiahDoNeumann** [comment]
Right. In this column quantity feature, I think the change is cool. I would only be in doubt about the name of the parameter.
1 - {{ $field['option_class'] ?? 'col-sm-4' }}
2 - {{ $field['column_class'] ?? 'col-sm-4' }}

What better name would represent the functionality or have another?

**Comment by tabacitu** [comment]
Eeeh... 🙄  I think `option_class` sounds a little better, it would pair well with the `options` attribute. I think it makes sense to define:
- the `options` that need to show in the 
- the class on each of the options with `option_class`

There's also the possible confusion with the Backpack Columns feature, that we'd be avoiding. Don't get me wrong, it makes _no sense_ to use Backpack Columns here, since we're inside a Field, but... it could be a source of confusion nonetheless. 

What do you think about the naming @pxpm ?

PS. I now see that the `options` attribute is NOT documented, so we should do that too:
```php
[
    'label'     => 'Roles',
    'type'      => 'checklist',
    'name'      => 'roles',
    'entity'    => 'roles',
    'attribute' => 'name',
    'model'     => "Backpack\PermissionManager\app\Models\Role",
    'pivot'     => true,
    
    // Optionals

    // Manually define the options in the checklist;
    // 'options' => (function ($query) {
    //     return $query->orderBy('name', 'ASC')->where('depth', 1)->get();
    // }),

    // Change how many options are shown per line 
    // 'option_class' => 'col-sm-4',
]
```
Hopefully now it seems more consistent and my argument above makes sense 😆

**Comment by dividy** [comment]
It happens I have the need for this feature too.

**option_class** sounds nice.
However, it seems this feature exists for _checklist_dependency_ and it is called : **number_columns** 

See: [](https://backpackforlaravel.com/docs/4.1/crud-fields#checklist_dependency)

**Comment by tabacitu** [comment]
Ouch. I now think `number_columns` is such a bad name for that... 😞  Come on past me, it could've at least been `number_of_columns`. But hey, they say "_If you're happy with the coding you've done a few years ago, it means you haven't evolved at all_". Soo... silver lining? 😆 

I think `option_class` will do just fine here, because it's more broad and it would also allow adding extra CSS classes, just in case (ex: maybe the developer wants a certain behaviour on SM, MD, LG and XL). Yes, it'll be different from `checklist_dependecy`, but... it's better soo... 😄 

Alternatively, we could do `number_of_columns`... But it wouldn't be as customizable.
